File: onboarding-servey/src/main/java/com/innercircle/onboardingservey/domain/model/entity/Answer.java
package com.innercircle.onboardingservey.domain.model.entity;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.NoArgsConstructor;

@Entity
@Table(name = "answer")
@NoArgsConstructor
@Getter
public class Answer {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long answerId;
    private Long userId;

    @ManyToOne(fetch = FetchType.LAZY)
    private SurveyVersion surveyVersion;

    public Answer(
        Long userId,
        SurveyVersion surveyVersion
    ) {
        this.userId = userId;
        this.surveyVersion = surveyVersion;
    }
}

File: src/main/kotlin/org/example/survey/domain/Answer.kt
package org.example.survey.domain

import jakarta.persistence.Entity
import jakarta.persistence.FetchType
import jakarta.persistence.GeneratedValue
import jakarta.persistence.GenerationType
import jakarta.persistence.Id
import jakarta.persistence.JoinColumn
import jakarta.persistence.ManyToOne
import jakarta.persistence.Table

@Entity
@Table(name = "answer")
class Answer(
    content: String,
    questionId: Long,
) {
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "feedback_id")
    var feedback: Feedback? = null

    val questionId: Long = questionId

    val content: String? = content

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    val answerId: Long = 0

    override fun toString(): String = "Answer(questionId=$questionId, content=$content, answerId=$answerId)"
}
